services/github/github-commit-activity.service.js

```diff
@@ -46,47 +52,67 @@ export default class GithubCommitActivity extends GithubAuthV3Service {
     }
   }
 
-  static transform({ interval, weekData }) {
-    const weekTotals = weekData.map(({ total }) => total)
+  async fetch({ interval, user, repo, branch = 'FETCH_HEAD' }) {
+    const since = this.constructor.getIntervalQueryStartDate({ interval })
+    return this._requestGraphql({
+      query: gql`
+        query (
+          $user: String!
+          $repo: String!
+          $branch: String!
+          $since: GitTimestamp!
+        ) {
+          repository(owner: $user, name: $repo) {
+            object(expression: $branch) {
```

Comment:
Best I can tell this value is used in a call to git's `rev-list` directly or indirectly via `rev-parse` and is a required part of the query. My understanding is that defaulting to this to `FETCH_HEAD` means it'll work with the default branch in the target repo regardless of what that branch is named, and seems like a good idea given we must provide something.

One thing I wasn't clear on was whether the existing v3 API did the same internally or if it spanned across all the branches. At the moment the v3 is still "down" (HTTP 202 - processing) so I can't really compare the results to check for any differences. I think if we go this route with this type of query then it makes sense to surface up the ref that's used here as a route parameter so that users can specify their preference.

Reply:
The docs are slightly unclear on this, but https://docs.github.com/en/rest/reference/repos#statistics says the stats match the graphs in the front end (e.g: https://github.com/badges/shields/graphs/contributors ). My interpretation is they do only show commits to the default branch. I think this assumption is good